### ISSUE TITLE
fatal error: runtime: C malloc failed

### ISSUE THREAD

**Issue by Pythonidas** [ethereum/go-ethereum#545]
### environment: Amazon EC2 instance AMI: ubuntu14.04 / Docker ubuntu:latest 

After fresh PPA ethereum install from ubuntu 14.04, after I type ethereum -mine, I gott this:

Note: I already created a key and used --unlock flag with corresponding address:passphrase.

===============================
root@ec2$:ethereum -mine

Welcome to the FRONTIER
2015/03/21 10:12:29 [SERV] Protocol Version: 58, Network Id: 0
2015/03/21 10:12:29 [CHAIN] Last block (#0) fd4af92a79c7fc2fd8bf0d342f2e832e1d4f485c85b9152d2039e03bc604fdca TD=0
2015/03/21 10:12:29 [POW] Making Cache
2015/03/21 10:12:30 [POW] Took: 1.263843747s
2015/03/21 10:12:30 [CLI] Starting  Ethereum(G)/v0.9.1/linux/go1.2.1
2015/03/21 10:12:30 [P2P Server] Starting Server
2015/03/21 10:12:31 [P2P Discovery] Listening,  enode://7c80422b2ca19d95a31ec88f387d32400052597bb795d7f412676b6af3fa44e6468d9ebeb61f805d6e671850585e07134f14dca075f0e2da4ff2ba7c504e4d99@[::]:30303
2015/03/21 10:12:31 [Blockpool] Started
2015/03/21 10:12:31 [P2P NAT] mapping error: no devices discovered
2015/03/21 10:12:31 [P2P Server] Listening on [::]:30303
2015/03/21 10:12:31 [SHH] Whisper started
2015/03/21 10:12:31 [P2P NAT] mapping error: no devices discovered
2015/03/21 10:12:31 [SERV] Server started
2015/03/21 10:12:32 [POW] Making Cache
2015/03/21 10:12:33 [POW] Took: 1.295639362s
2015/03/21 10:12:33 [POW] Retrieving DAG
fatal error: runtime: C malloc failed

goroutine 1 [running]:
runtime.throw(0x14526e8)
        /usr/lib/go/src/pkg/runtime/panic.c:464 +0x69 fp=0x7f1b97c56820
runtime.cmalloc(0x3ffff880)
        /usr/lib/go/src/pkg/runtime/cgocall.c:202 +0x5d fp=0x7f1b97c56848
github.com/ethereum/ethash._Cfunc__CMalloc(0x3ffff880, 0x3)
        github.com/ethereum/ethash/_obj/_cgo_defun.c:40 +0x27 fp=0x7f1b97c56858
github.com/ethereum/ethash.makeDAG(0xc210e09960, 0xb94ad0)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/ethereum/ethash/ethash.go:121 +0x2d fp=0x7f1b97c56898
github.com/ethereum/ethash.(*Ethash).UpdateDAG(0xc210d31d70)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/ethereum/ethash/ethash.go:202 +0x5b9 fp=0x7f1b97c56ab0
github.com/ethereum/go-ethereum/miner.(*Miner).Start(0xc210d31dc0, 0xc210e09940, 0x14, 0x14)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/miner/miner.go:40 +0x136 fp=0x7f1b97c56b08
github.com/ethereum/go-ethereum/eth.(*Ethereum).StartMining(0xc2100bd0f0, 0xaaf100, 0x4)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/eth/backend.go:298 +0x210 fp=0x7f1b97c56b98
main.startEth(0xc210070300, 0xc2100bd0f0)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/cmd/ethereum/main.go:243 +0x2a5 fp=0x7f1b97c56c30
main.run(0xc210070300)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/cmd/ethereum/main.go:184 +0x163 fp=0x7f1b97c56cb0
github.com/codegangsta/cli.(*App).Run(0xc2100bd000, 0xc21000a000, 0x3, 0x3, 0x0, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:180 +0xbd7 fp=0x7f1b97c56eb8
main.main()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/cmd/ethereum/main.go:169 +0x82 fp=0x7f1b97c56f48
runtime.main()
        /usr/lib/go/src/pkg/runtime/proc.c:220 +0x11f fp=0x7f1b97c56fa0
runtime.goexit()
        /usr/lib/go/src/pkg/runtime/proc.c:1394 fp=0x7f1b97c56fa8

goroutine 3 [runnable]:
github.com/ethereum/go-ethereum/logger.dispatchLoop()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/logger/sys.go:41 +0x512
created by github.com/ethereum/go-ethereum/logger.init·1
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/logger/sys.go:20 +0x21

goroutine 4 [syscall]:
runtime.goexit()
        /usr/lib/go/src/pkg/runtime/proc.c:1394

goroutine 5 [syscall]:
os/signal.loop()
        /usr/lib/go/src/pkg/os/signal/signal_unix.go:21 +0x1e
created by os/signal.init·1
        /usr/lib/go/src/pkg/os/signal/signal_unix.go:27 +0x31

goroutine 6 [chan receive]:
github.com/ethereum/go-ethereum/cmd/utils.func·001()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:51 +0xd0
created by github.com/ethereum/go-ethereum/cmd/utils.HandleInterrupt
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:55 +0x7f

goroutine 30 [select]:
github.com/ethereum/go-ethereum/p2p/discover.(*udp).loop(0xc210edc8c0)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:239 +0x89c
created by github.com/ethereum/go-ethereum/p2p/discover.ListenUDP
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:155 +0x41d

goroutine 31 [IO wait]:
net.runtime_pollWait(0x7f1b97df7c68, 0x72, 0x0)
        /usr/lib/go/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc210ed1df0, 0x72, 0x7f1b97df6240, 0xb)
        /usr/lib/go/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc210ed1df0, 0xb, 0x7f1b97df6240)
        /usr/lib/go/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).ReadFrom(0xc210ed1d90, 0xc210eeb000, 0x1000, 0x1000, 0x0, ...)
        /usr/lib/go/src/pkg/net/fd_unix.go:231 +0x320
net.(*UDPConn).ReadFromUDP(0xc210d4c5b8, 0xc210eeb000, 0x1000, 0x1000, 0x1000, ...)
        /usr/lib/go/src/pkg/net/udpsock_posix.go:67 +0x122
github.com/ethereum/go-ethereum/p2p/discover.(*udp).readLoop(0xc210edc8c0)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:324 +0xd3
created by github.com/ethereum/go-ethereum/p2p/discover.ListenUDP
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:156 +0x437

goroutine 28 [select]:
github.com/ethereum/go-ethereum/event/filter.(*Filters).loop(0xc210d62a00)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/event/filter/filter.go:57 +0x22d
created by github.com/ethereum/go-ethereum/event/filter.(*Filters).Start
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/event/filter/filter.go:32 +0x2e

goroutine 10 [chan receive]:
github.com/ethereum/go-ethereum/logger.sysLoop(0x7f1b97df6a20, 0xc2100c1520, 0xc2100d7000, 0xc210050880)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/logger/sys.go:77 +0x3d
created by github.com/ethereum/go-ethereum/logger.func·001
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/logger/sys.go:37 +0x12d

goroutine 33 [select]:
github.com/ethereum/go-ethereum/p2p/nat.Map(0x7f1b97df69e0, 0xc2100b5ba0, 0xc210ed2b40, 0xab6e80, 0x3, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/nat/nat.go:98 +0x74e
github.com/ethereum/go-ethereum/p2p.func·005()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:222 +0x90
created by github.com/ethereum/go-ethereum/p2p.(*Server).startListening
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:224 +0x212

goroutine 32 [IO wait]:
net.runtime_pollWait(0x7f1b97df7db8, 0x72, 0x0)
        /usr/lib/go/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc210ee1140, 0x72, 0x7f1b97df6240, 0xb)
        /usr/lib/go/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc210ee1140, 0xb, 0x7f1b97df6240)
        /usr/lib/go/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).accept(0xc210ee10e0, 0xbd60d8, 0x0, 0x7f1b97df6240, 0xb)
        /usr/lib/go/src/pkg/net/fd_unix.go:382 +0x2c2
net.(*TCPListener).AcceptTCP(0xc210d4c610, 0x7f1b00000003, 0x7f1b97c62f70, 0x2)
        /usr/lib/go/src/pkg/net/tcpsock_posix.go:233 +0x47
net.(*TCPListener).Accept(0xc210d4c610, 0x7f1b97c62f70, 0x2, 0x2, 0xc21007ed80)
        /usr/lib/go/src/pkg/net/tcpsock_posix.go:243 +0x27
github.com/ethereum/go-ethereum/p2p.(*Server).listenLoop(0xc210d54b40)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:263 +0x189
created by github.com/ethereum/go-ethereum/p2p.(*Server).startListening
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:218 +0x17a

goroutine 13 [IO wait]:
net.runtime_pollWait(0x7f1b97df7d10, 0x72, 0x0)
        /usr/lib/go/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc2100ad450, 0x72, 0x7f1b97df6240, 0xb)
        /usr/lib/go/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc2100ad450, 0xb, 0x7f1b97df6240)
        /usr/lib/go/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).ReadFrom(0xc2100ad3f0, 0xc2100c1830, 0xc, 0xc, 0x0, ...)
        /usr/lib/go/src/pkg/net/fd_unix.go:231 +0x320
net.(*UDPConn).ReadFromUDP(0xc210000a20, 0xc2100c1830, 0xc, 0xc, 0x2, ...)
        /usr/lib/go/src/pkg/net/udpsock_posix.go:67 +0x122
github.com/jackpal/go-nat-pmp.(*Client).rpc(0xc2100c55a0, 0xc210000a00, 0x2, 0x2, 0xc, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/jackpal/go-nat-pmp/natpmp.go:126 +0x2ea
github.com/jackpal/go-nat-pmp.(*Client).GetExternalAddress(0xc2100c55a0, 0x0, 0x0, 0x0)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/jackpal/go-nat-pmp/natpmp.go:50 +0xbf
github.com/ethereum/go-ethereum/p2p/nat.func·006()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/nat/natpmp.go:57 +0x7f
created by github.com/ethereum/go-ethereum/p2p/nat.discoverPMP
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/nat/natpmp.go:62 +0x125

goroutine 15 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc210075900)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:111 +0x1c5
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:116 +0x465

goroutine 16 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc210075900)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:666 +0x6d7
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:119 +0x4a6

goroutine 17 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc210075900)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:615 +0x199
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:120 +0x4bd

goroutine 18 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc210075900)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x13d
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:121 +0x4d4

goroutine 19 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc210075b40)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:111 +0x1c5
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:116 +0x465

goroutine 20 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc210075b40)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:666 +0x6d7
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:119 +0x4a6

goroutine 21 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc210075b40)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:615 +0x199
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:120 +0x4bd

goroutine 22 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc210075b40)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x13d
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:121 +0x4d4

goroutine 23 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc210075d80)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:111 +0x1c5
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:116 +0x465

goroutine 24 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc210075d80)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:666 +0x6d7
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:119 +0x4a6

goroutine 25 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc210075d80)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:615 +0x199
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:120 +0x4bd

goroutine 26 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc210075d80)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x13d
created by github.com/syndtr/goleveldb/leveldb.openDB
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:121 +0x4d4

goroutine 27 [select]:
github.com/ethereum/go-ethereum/core.(*ChainManager).update(0xc210d3d420)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/core/chain_manager.go:479 +0x4d9
created by github.com/ethereum/go-ethereum/core.NewChainManager
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/core/chain_manager.go:103 +0x199

goroutine 29 [select]:
github.com/ethereum/go-ethereum/p2p/nat.Map(0x7f1b97df69e0, 0xc2100b5ba0, 0xc210ed2c00, 0xab81a0, 0x3, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/nat/nat.go:98 +0x74e
created by github.com/ethereum/go-ethereum/p2p/discover.ListenUDP
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:146 +0x2bb

goroutine 34 [select]:
github.com/ethereum/go-ethereum/p2p.(*Server).dialLoop(0xc210d54b40)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:289 +0x554
created by github.com/ethereum/go-ethereum/p2p.(*Server).Start
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:199 +0x746

goroutine 35 [select]:
github.com/ethereum/go-ethereum/blockpool.func·001()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/blockpool/blockpool.go:198 +0x16d
created by github.com/ethereum/go-ethereum/blockpool.(*BlockPool).Start
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/blockpool/blockpool.go:205 +0x35f

goroutine 60 [chan receive]:
github.com/ethereum/go-ethereum/p2p/discover.(*udp).findnode(0xc210edc8c0, 0xc210ed3200, 0x959da12c2b42807c, 0x40327d388fc81ea3, 0xf4d795b77b595200, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:199 +0x21c
github.com/ethereum/go-ethereum/p2p/discover.func·001()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:110 +0x6c
created by github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:112 +0x43e

goroutine 53 [chan receive]:
github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup(0xc210091100, 0x959da12c2b42807c, 0x40327d388fc81ea3, 0xf4d795b77b595200, 0xe644faf36a6b6712, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:121 +0x4c0
github.com/ethereum/go-ethereum/p2p.(*Server).findPeers(0xc210d54b40)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:344 +0xf3
created by github.com/ethereum/go-ethereum/p2p.(*Server).dialLoop
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/server.go:279 +0xfb

goroutine 38 [select]:
github.com/ethereum/go-ethereum/whisper.(*Whisper).update(0xc210d5fa10)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/whisper/whisper.go:222 +0xbf
created by github.com/ethereum/go-ethereum/whisper.(*Whisper).Start
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/whisper/whisper.go:89 +0xc9

goroutine 39 [chan receive]:
github.com/ethereum/go-ethereum/eth.(*Ethereum).txBroadcastLoop(0xc2100bd0f0)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/eth/backend.go:406 +0x57
created by github.com/ethereum/go-ethereum/eth.(*Ethereum).Start
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/eth/backend.go:347 +0x200

goroutine 40 [chan receive]:
github.com/ethereum/go-ethereum/eth.(*Ethereum).blockBroadcastLoop(0xc2100bd0f0)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/eth/backend.go:414 +0x62
created by github.com/ethereum/go-ethereum/eth.(*Ethereum).Start
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/eth/backend.go:351 +0x2d4

goroutine 61 [chan receive]:
github.com/ethereum/go-ethereum/p2p/discover.(*udp).findnode(0xc210edc8c0, 0xc210ed3280, 0x959da12c2b42807c, 0x40327d388fc81ea3, 0xf4d795b77b595200, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:199 +0x21c
github.com/ethereum/go-ethereum/p2p/discover.func·001()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:110 +0x6c
created by github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:112 +0x43e

goroutine 62 [chan receive]:
github.com/ethereum/go-ethereum/p2p/discover.(*udp).findnode(0xc210edc8c0, 0xc210ed3300, 0x959da12c2b42807c, 0x40327d388fc81ea3, 0xf4d795b77b595200, ...)
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:199 +0x21c
github.com/ethereum/go-ethereum/p2p/discover.func·001()
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:110 +0x6c
created by github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup
        /build/buildd/ethereum-0.9.1+232SNAPSHOT20150321062918trusty/obj-x86_64-linux-gnu/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:112 +0x43e

**Comment by obscuren** [comment]
You ran out of memory? :-) ASIC resistance ftw

**Comment by Pythonidas** [comment]
Well, it definitely shot down my 1GB t2micro...thanks for explaining!

**Comment by obscuren** [comment]
No problem. I'll also move the dag generation out of `/tmp`. Someone told me that `/tmp` is usually less than 2gb

**Comment by zelig** [comment]
gav: https://github.com/ethereum/wiki/wiki/Ethash-DAG for the standard on where the DAG should be stored on the system

**Comment by CurreE** [comment]
I am getting this exact same problem, and it's got me completely stumped.  This is using the latest CLI that I just built today.  My environment is Windows 8 64-bit, with 8 GB RAM and more than 1 TB of free disk space.  So I definitely should not be experiencing any out-of-memory issues.  Any idea how to get around this?

**Comment by Pythonidas** [comment]
You think there is need to reopen this issue?

**Comment by Pythonidas** [comment]
You think there is need to reopen this issue?

**Comment by CurreE** [comment]
I'm not sure if this should be re-opened, or if I should just make a new ticket since my operating environment is very different from yours.

May I ask why you closed this ticket?  Is the program working for you now?

**Comment by obscuren** [comment]
Because:

> Well, it definitely shot down my 1GB t2micro...thanks for explaining!

You'll clearly run out of mem (so it's a non issue you are having)

Regarding windows: https://github.com/ethereum/go-ethereum/issues/524

**Comment by CurreE** [comment]
Okay, got it, thanks.  So for Pythonidas there was a reasonable explanation and running out of memory would not be an unexpected event.  But as I mentioned, my Windows machine has 8 GB RAM (and according to task manager there was more than 5 GB free when I tested ethereum), so memory should not be a concern.  I tried it in a Git Bash shell and also in a regular DOS prompt thinking maybe the bash shell had access to only limited system resources for some reason, but got the same result.

I took a look at #524 but that seems to be a different error.  Unless you think it's related, or want to keep track of all Windows issues on the same ticket.
